voucher/src/main/java/com/adyen/checkout/voucher/internal/ui/view/FullVoucherView.kt

```diff
@@ -186,6 +193,35 @@ internal class FullVoucherView @JvmOverloads constructor(
         }
     }
 
+    private fun updateReadInstructionTextView(instructionUrl: String?) {
+        binding.textViewReadInstructions.isVisible = instructionUrl != null
+        instructionUrl?.let {
+            binding.textViewReadInstructions.text =
+                localizedContext.getString(R.string.checkout_voucher_read_instructions).formatFullStringWithHyperLink()
+            binding.textViewReadInstructions.setOnClickListener {
+                onReadInstructionsClicked(instructionUrl)
+            }
+        }
+    }
+
+    private fun onReadInstructionsClicked(url: String) {
```

Comment:
If this will always be a PDF, could you use the `PdfOpener.open()` instead?

Reply:
If you use `PdfOpener`, the logic to handle `onReadInstructionsClicked` could move to the `DefaultVoucherDelegate` and the `instructionUrl` could be removed from `VoucherOutputData`.